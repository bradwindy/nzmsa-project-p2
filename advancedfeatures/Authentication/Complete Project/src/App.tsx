import * as React from 'react';
import ReactPlayer from 'react-player';
import CaptionArea from 'src/Components/CaptionArea';
import Header from 'src/Components/Header';
import VideoList from 'src/Components/VideoList';
import 'src/App.css';
import * as Webcam from "react-webcam";

interface IState {
    updateVideoList: any,
    player: any,
    playingURL: string
    videoList: object,
    authenticated: boolean,
    refCamera: any,
    predictionResult: any
}

class App extends React.Component<{}, IState>{
    public constructor(props: any) {
        super(props);
        this.state = {
            player: null,
            playingURL: "",
            updateVideoList: null,
            videoList: [],
            authenticated: false,
            refCamera: React.createRef(),
            predictionResult: null
        }

        this.authenticate = this.authenticate.bind(this)

    }

    public setRef = (playerRef: any) => {
        this.setState({
            player: playerRef
        })
    }

    public addVideo = (url: string) => {
        const body = { "url": url }
        fetch("https://scriberapi.azurewebsites.net/api/Videos", {
            body: JSON.stringify(body),
            headers: {
                Accept: "text/plain",
                "Content-Type": "application/json"
            },
            method: "POST"
        }).then(() => {
            this.state.updateVideoList();
        })
    }

    public updateURL = (url: string) => {
        if (this.state.playingURL === url) {
            this.setState({ playingURL: "" }, () => this.setState({ playingURL: url }))
        } else {
            this.setState({ playingURL: url })
        }
    }

    public listMounted = (callbacks: any) => {
        this.setState({ updateVideoList: callbacks })
    }

    public render() {

        const { authenticated } = this.state

        return (<div>

            <div>
                {(!authenticated) ?
                    <div>
                        <Webcam
                            audio={false}
                            screenshotFormat="image/jpeg"
                            ref={this.state.refCamera}
                        />
                        <div className="row nav-row">
                            <div className="btn btn-primary bottom-button" onClick={this.authenticate}>Login</div>
                        </div>
                    </div>
                    : ""}


                {(authenticated) ?
                    <div>
                        <Header addVideo={this.addVideo} />
                        <div className="container">
                            <div className="row">
                                <div className="col-7">
                                    <ReactPlayer
                                        className="player"
                                        ref={this.setRef}
                                        controls={true}
                                        url={this.state.playingURL}
                                        width="100%"
                                        height="400px"
                                        playing={true}
                                        config={{
                                            youtube: {
                                                playerVars: { showinfo: 1 },
                                                preload: true
                                            }
                                        }
                                        }
                                    />
                                </div>
                                <div className="col-5">
                                    <VideoList play={this.updateURL} mount={this.listMounted} />
                                </div>
                            </div>
                            <CaptionArea currentVideo={this.state.playingURL} play={this.updateURL} />
                        </div>
                    </div>
                    : ""}

            </div>

        </div>)
    }

    // Call custom vision model
    private getFaceRecognitionResult(image: string) {
        const url = "[API-ENDPOINT]"
        if (image === null) {
            return;
        }
        const base64 = require('base64-js');
        const base64content = image.split(";")[1].split(",")[1]
        const byteArray = base64.toByteArray(base64content);
        fetch(url, {
            body: byteArray,
            headers: {
                'cache-control': 'no-cache', 'Prediction-Key': '[API-KEY]', 'Content-Type': 'application/octet-stream'
            },
            method: 'POST'
        })
            .then((response: any) => {
                if (!response.ok) {
                    // Error State
                    alert(response.statusText)
                } else {
                    response.json().then((json: any) => {
                        console.log(json.predictions[0])

                        this.setState({ predictionResult: json.predictions[0] })
                        if (this.state.predictionResult.probability > 0.7) {
                            this.setState({ authenticated: true })
                        } else {
                            this.setState({ authenticated: false })
                            console.log(json.predictions[0].tagName)
                        }
                    })
                }
            })
    }

    // Authenticate
    private authenticate() {
        const screenshot = this.state.refCamera.current.getScreenshot();
        this.getFaceRecognitionResult(screenshot);
    }

}

export default App;