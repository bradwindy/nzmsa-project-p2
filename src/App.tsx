import * as React from "react";
import ReactPlayer from "react-player";
import CaptionArea from "./Components/CaptionArea";
import Header from "./Components/Header";
import VideoList from "./Components/VideoList";
import "./App.css";
import { library } from "@fortawesome/fontawesome-svg-core";
import {
  faHeart as fasHeart,
  faPlus,
  faPlusCircle,
  faPlusSquare,
  faStar as fasStar,
  faTimes,
} from "@fortawesome/free-solid-svg-icons";
import {
  faHeart as farHeart,
  faStar as farStar,
} from "@fortawesome/free-regular-svg-icons";

library.add(
  fasStar,
  farStar,
  fasHeart,
  farHeart,
  faPlusCircle,
  faPlus,
  faPlusSquare,
  faTimes,
);

interface IState {
  playingUrl: any;
  updateVideoList: any;
}

class App extends React.Component<{}, IState> {
  public constructor(props: any) {
    super(props);
    this.state = {
      playingUrl: "",
      updateVideoList: null,
    };
  }

  public addVideo = (url: any) => {
    const body = { url };
    fetch("https://scriberapi.azurewebsites.net/api/Videos", {
      body: JSON.stringify(body),
      headers: {
        "Accept": "text/plain",
        "Content-Type": "application/json",
      },
      method: "POST",
    }).then(() => {
      this.state.updateVideoList();
    });
  };

  public updateURL = (url: string) => {
    if (this.state.playingUrl === url) {
      this.setState({ playingUrl: "" }, () =>
        this.setState({ playingUrl: url }),
      );
    } else {
      this.setState({ playingUrl: url });
    }
  };

  public videoList = (callback: any) => {
    this.setState({ updateVideoList: callback });
  };

  public render() {
    // @ts-ignore
    return (
      <div>
        <Header addVideo={this.addVideo} />
        <div className="container">
          <div className="row">
            <div className="col-7">
              {/*
              // @ts-ignore */}
              <ReactPlayer
                className="player"
                controls={true}
                url={this.state.playingUrl}
                width="100%"
                height="400px"
                playing={true}
                config={{
                  youtube: {
                    playerVars: { showinfo: 1 },
                    preload: true,
                  },
                }}
              />
            </div>
            <div className="col-5">
              <VideoList play={this.updateURL} mount={this.videoList} />
            </div>
          </div>
          <CaptionArea
            play={this.updateURL}
            currentVideo={this.state.playingUrl}
          />
        </div>
      </div>
    );
  }
}

export default App;
