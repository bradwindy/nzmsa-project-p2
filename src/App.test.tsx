import * as React from "react";
import * as ReactDOM from "react-dom";
import App from "./App";
import { create } from "react-test-renderer";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe("App Button Match", () => {
  test("Testing app button", () => {
    const tree = create(<App />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});