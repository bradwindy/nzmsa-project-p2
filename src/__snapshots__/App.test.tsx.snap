// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Button Match Testing app button 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="col"
    />
    <div
      className="header p-0 m-0 mb-3 col-12"
    >
      <nav
        className="navbar navbar-light pt-4 pb-0 pl-0 pr-0"
      >
        <h2
          className="float-left p-0 m-0"
          id="logotext"
        >
          <img
            alt=""
            height="30"
            src="logo.svg"
            width="30"
          />
          captiv
        </h2>
        <div
          className="col-6 float-right p-0"
        >
          <div
            className="form-inline float-right"
          >
            <input
              aria-label="Search"
              className="form-control SearchBar border-light col float-right"
              id="Search-Bar"
              onChange={[Function]}
              placeholder="Add Video URL"
              type="search"
            />
            <button
              className="btn btn-outline-success btn-outline-light float-right ml-2"
              onClick={[Function]}
            >
              + Add
            </button>
          </div>
        </div>
      </nav>
    </div>
    <div
      className="col"
    />
  </div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-7 pt-3"
      >
        <div
          className="player"
          style={
            Object {
              "height": "400px",
              "width": "100%",
            }
          }
        >
          <div
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
          >
            <div />
          </div>
        </div>
      </div>
      <div
        className="col-lg-5 pt-3"
      >
        <div
          className="video-list card border-light"
        >
          <h4
            className="pt-3 pl-3 pb-2"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-film fa-w-16 fa-sm "
              data-icon="film"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={
                Object {
                  "color": "white",
                }
              }
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M488 64h-8v20c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12V64H96v20c0 6.6-5.4 12-12 12H44c-6.6 0-12-5.4-12-12V64h-8C10.7 64 0 74.7 0 88v336c0 13.3 10.7 24 24 24h8v-20c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v20h320v-20c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v20h8c13.3 0 24-10.7 24-24V88c0-13.3-10.7-24-24-24zM96 372c0 6.6-5.4 12-12 12H44c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40zm0-96c0 6.6-5.4 12-12 12H44c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40zm0-96c0 6.6-5.4 12-12 12H44c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40zm272 208c0 6.6-5.4 12-12 12H156c-6.6 0-12-5.4-12-12v-96c0-6.6 5.4-12 12-12h200c6.6 0 12 5.4 12 12v96zm0-168c0 6.6-5.4 12-12 12H156c-6.6 0-12-5.4-12-12v-96c0-6.6 5.4-12 12-12h200c6.6 0 12 5.4 12 12v96zm112 152c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40zm0-96c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40zm0-96c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
             
            videos
          </h4>
          <table
            className="table"
          />
        </div>
      </div>
    </div>
    <div
      className="caption-area card mt-4 p-4 border-light mb-5"
    >
      <div
        className="row"
      >
        <div
          className="col"
        >
          <h4
            className="font-weight-bold"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-closed-captioning fa-w-16 fa-sm "
              data-icon="closed-captioning"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={
                Object {
                  "color": "white",
                }
              }
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M464 64H48C21.5 64 0 85.5 0 112v288c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V112c0-26.5-21.5-48-48-48zM218.1 287.7c2.8-2.5 7.1-2.1 9.2.9l19.5 27.7c1.7 2.4 1.5 5.6-.5 7.7-53.6 56.8-172.8 32.1-172.8-67.9 0-97.3 121.7-119.5 172.5-70.1 2.1 2 2.5 3.2 1 5.7l-17.5 30.5c-1.9 3.1-6.2 4-9.1 1.7-40.8-32-94.6-14.9-94.6 31.2.1 48 51.1 70.5 92.3 32.6zm190.4 0c2.8-2.5 7.1-2.1 9.2.9l19.5 27.7c1.7 2.4 1.5 5.6-.5 7.7-53.5 56.9-172.7 32.1-172.7-67.9 0-97.3 121.7-119.5 172.5-70.1 2.1 2 2.5 3.2 1 5.7L420 222.2c-1.9 3.1-6.2 4-9.1 1.7-40.8-32-94.6-14.9-94.6 31.2 0 48 51 70.5 92.2 32.6z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
             
            search captions
          </h4>
        </div>
        <div
          className="col-6 float-right p-0"
        >
          <div
            className="form-inline float-right mr-2"
          >
            <input
              aria-label="Search"
              className="form-control SearchBar border-light col"
              id="Search-Bar"
              onChange={[Function]}
              placeholder="Search"
              type="search"
              value=""
            />
            <button
              className="btn btn-outline-success my-2 my-sm-0 btn-outline-light ml-2"
              onClick={[Function]}
            >
              Search
            </button>
          </div>
        </div>
      </div>
      <br />
      <table
        className="table"
      >
        <tbody>
          <tr
            className="p-2 text-white"
          >
            <th
              className="p-2 font-weight-bold"
            >
              Time
            </th>
            <th
              className="p-2 font-weight-bold"
            >
              Caption
            </th>
            <th
              className="p-2 font-weight-bold"
            >
              Video
            </th>
          </tr>
        </tbody>
        <tbody
          className="captionTable p-2"
        />
      </table>
    </div>
  </div>
</div>
`;
